menas/src/main/scala/za/co/absa/enceladus/rest/services/VersionedModelService.scala

```diff
@@ -52,28 +55,48 @@ abstract class VersionedModelService[C <: VersionedModel](versionedMongoReposito
 
   def getUsedIn(name: String, version: Option[Int]): Future[UsedIn]
 
-  def create(item: C, username: String): Future[Option[C]] = {
+  private[rest] def getMenasRef(item: C): MenasReference = MenasReference(Some(versionedMongoRepository.collectionName), item.name, item.version)
+
+  private[services] def getAuditEntry(item: C, entryType: AuditEntryType, username: String, auditMessage: String): AuditEntry = {
+    AuditEntry(menasRef = Some(getMenasRef(item)), entryType = entryType, user = username, timeCreated = ZonedDateTime.now(), message = auditMessage)
+  }
+  
+  private[services] def getAuditEntry(itemName: String, itemVersion: Int, entryType: AuditEntryType, username: String, auditMessage: String): AuditEntry = {
+    AuditEntry(menasRef = Some(MenasReference(Some(versionedMongoRepository.collectionName), itemName, itemVersion)), 
+        entryType = entryType, user = username, timeCreated = ZonedDateTime.now(), message = auditMessage)
+  }
+
+  def create(item: C, username: String): Future[Option[C]]
+  
+  private[services] def create(item: C, username: String, auditMessage: String): Future[Option[C]] = {
     for {
       validation <- validate(item)
-      _ <-
+      _          <-
         if (validation.isValid()) versionedMongoRepository.create(item, username)
         else throw ValidationException(validation)
+      audit <- auditTrailService.create(getAuditEntry(item, CreateEntryType, username, auditMessage))
       detail <- getLatestVersion(item.name)
     } yield detail
   }
-
+  
   def update(username: String, item: C): Future[Option[C]]
 
-  def update(username: String, itemName: String)(transform: C => C): Future[Option[C]] = {
-    getLatestVersion(itemName).flatMap {
-      case Some(latest) =>
-        versionedMongoRepository.update(username, transform(latest)).flatMap { _ =>
-          getLatestVersion(itemName)
-        }
-      case None => Future.successful(None)
-    }
+  private[services] def update(username: String, itemName: String, itemVersion: Int, auditMessage: String)(transform: C => ChangedFieldsUpdateTransformResult[C]): Future[Option[C]] = {
+    for {
+      version <- getVersion(itemName, itemVersion)
+      transformed <- if(version.isEmpty) Future.failed(NotFoundException(s"Version $itemVersion of $itemName not found"))
+                      else Future.successful(transform(version.get))
+      update <- versionedMongoRepository.update(username, transformed.updatedEntity)
```

Comment:
Style - when you are aligning other `<-`, you could these too

Reply:
yea... not currently supported by scalariform.. Not gonna maintain this manually